class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
    

        for(int i =0;i<nums.length;i++){
            if(map.containsKey(nums[i])){
                int val = map.get(nums[i]);
                val += 1;

                map.put(nums[i],val);
            }else{
                map.put(nums[i],1);
            }
        }
        List<Integer> arr = new ArrayList<>();
        for(int key : map.keySet()){
            if(map.get(key)== 2){
                arr.add(key);
            }
        }
        return arr;
    }
}